nsx/config.go

```diff
@@ -0,0 +1,82 @@
+package nsx
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/IBM-tfproviders/govnsx"
+)
+
+type Config struct {
+	User          string
+	Password      string
+	NsxManagerUri string
+	InsecureFlag  bool
+	Debug         bool
+	DebugPath     string
+	DebugPathRun  string
+}
+
+//
+// Client() returns a new client for accessing NSX manager
+//
+func (c *Config) Client() (*govnsx.Client, error) {
+
+	err := c.EnableDebug()
+	if err != nil {
+		return nil, fmt.Errorf("Error setting up client debug: %s", err)
+	}
+
+	nsxMrgParams := &govnsx.NsxManagerParams{
+		UserName:      c.User,
+		Password:      c.Password,
+		Uri:           c.NsxManagerUri,
+		AllowInsecssl: c.InsecureFlag,
+		RestAgentName: "govnsx",
```

Comment:
RestAgentName should be configurable with some default value "Terraform-Nsx-Provider"